src/seedu/addressbook/commands/order/DraftOrderEditPointsCommand.java

```diff
@@ -33,14 +32,20 @@ public DraftOrderEditPointsCommand(int points) {
     @Override
     public CommandResult execute() {
         try {
-            final ReadOnlyMember member = rms.getMemberFromDraftOrder();
-            final int points = toRedeem.getPoints();
-            final ReadOnlyMember emptyMember = new Member();
-            if (member.getName().equals(emptyMember.getName())) {
+            final ReadOnlyOrder draftOrder = rms.getDraftOrder();
+            final ReadOnlyMember customer = draftOrder.getCustomer();
+            int points = toRedeem.getPoints();
+            if (!draftOrder.hasCustomerField()) {
                 throw new IllegalValueException("Member needs to be added first!");
-            } else if (member.getPointsValue() < points) {
+            } else if (customer.getPointsValue() < points) {
                 throw new IllegalValueException("Member does not have sufficient points to redeem!");
+            } else if (points < 0) {
+                throw new IllegalValueException("Points to be redeemed must not be a negative value");
             } else {
+                int maxPointsRedeemable = draftOrder.getMaxPointsRedeemable();
+                if (points > maxPointsRedeemable) {
+                    points = maxPointsRedeemable;
```

Comment:
i don't think you should automatically set points to the maximum points that can be redeemed. Rather, a message should be displayed on the screen to tell the user that you have exceeded the total amount of points that can be redeemed.

Reply:
i'm just afraid that it might be viewed as a bug. Anyway the user managing the system is gonna be the cashier, so the cashier should be able to see the available points to be redeemed. So when a customer comes in to make an order and wants to use all his points, the cashier just has to key in the number seen in the available points field.

Reply:
i am going to shows the maximum amount of points that can be added like this then:
Redeemed points: 0 / 144
I think that I will set the implementation such that the limit is set as the min(user limit, order limit) and keep the implementation of all overflow is set back to limit.
This also means that I will remove the error message of points overflow the customer limits as this implementation prevents that from ever happen

Reply:
I think less error message and error cases = less hassle for users

Reply:
Yeah, that is what I said: limit = min(member limit, order limit)

Reply:
ok